train_segmentation.py

```diff
@@ -79,51 +79,53 @@ def create_dataloader(samples_folder: Path,
-    logging.info(f"Number of samples : {num_samples}\n")
+        raise logging.critical(ValueError(f"\nNumber of samples in .hdf5 files is less than batch size"))
+    logging.info(f"\nNumber of samples : {num_samples}")
     if not meta_map:
         dataset_constr = create_dataset.SegmentationDataset
     else:
         dataset_constr = functools.partial(create_dataset.MetaSegmentationDataset, meta_map=meta_map)
     datasets = []
 
     for subset in ["trn", "val", "tst"]:
-
         datasets.append(dataset_constr(samples_folder, subset, num_bands,
                                        max_sample_count=num_samples[subset],
                                        dontcare=dontcare_val,
-                                       radiom_transform=aug.compose_transforms(params=params,
+                                       radiom_transform=aug.compose_transforms(params=cfg,
                                                                                dataset=subset,
                                                                                aug_type='radiometric'),
-                                       geom_transform=aug.compose_transforms(params=params,
+                                       geom_transform=aug.compose_transforms(params=cfg,
                                                                              dataset=subset,
                                                                              aug_type='geometric',
                                                                              dontcare=dontcare_val,
                                                                              crop_size=crop_size),
-                                       totensor_transform=aug.compose_transforms(params=params,
+                                       totensor_transform=aug.compose_transforms(params=cfg,
                                                                                  dataset=subset,
                                                                                  input_space=BGR_to_RGB,
                                                                                  scale=scale,
                                                                                  dontcare2backgr=dontcare2backgr,
                                                                                  dontcare=dontcare_val,
                                                                                  aug_type='totensor'),
-                                       params=params,
+                                       params=cfg,
                                        debug=debug))
     trn_dataset, val_dataset, tst_dataset = datasets
 
-    # https://discuss.pytorch.org/t/guidelines-for-assigning-num-workers-to-dataloader/813/5
-    num_workers = len(gpu_devices_dict.keys()) * 4 if len(gpu_devices_dict.keys()) > 1 else 4
+    # Number of workers
+    if cfg.trainer.num_workers:
+        num_workers = cfg.trainer.num_workers
```

Comment:
The value for num_workers in this cfg yaml is 10. Isn't this more or less hardcoding the num_workers to 10? If we don't want to use this value, we'd have to explicitly overwrite trainer.num_workers to None?

Reply:
Yes, but good point, i will change the default value to None